src/esp/physics/PhysicsManager.h

```diff
@@ -0,0 +1,211 @@
+  void setRotation(const int physObjectID,
+                   const Magnum::Quaternion& quaternion);
+  void resetTransformation(const int physObjectID);
+
+  void translate(const int physObjectID, const Magnum::Vector3& vector);
+  void translateLocal(const int physObjectID, const Magnum::Vector3& vector);
+
+  void rotate(const int physObjectID,
+              const Magnum::Rad angleInRad,
+              const Magnum::Vector3& normalizedAxis);
+  void rotateX(const int physObjectID, const Magnum::Rad angleInRad);
+  void rotateY(const int physObjectID, const Magnum::Rad angleInRad);
+  void rotateZ(const int physObjectID, const Magnum::Rad angleInRad);
+  void rotateXLocal(const int physObjectID, const Magnum::Rad angleInRad);
+  void rotateYLocal(const int physObjectID, const Magnum::Rad angleInRad);
+  void rotateZLocal(const int physObjectID, const Magnum::Rad angleInRad);
+
+  const Magnum::Matrix4 getTransformation(const int physObjectID);
+  const Magnum::Vector3 getTranslation(const int physObjectID);
+  const Magnum::Quaternion getRotation(const int physObjectID);
+
+  // ============ Object Setter functions =============
+  // Setters that interface with physics need to take
+  void setMass(const int physObjectID, const double mass);
+  void setCOM(const int physObjectID, const Magnum::Vector3& COM);
+  void setInertia(const int physObjectID, const Magnum::Vector3& inertia);
+  void setScale(const int physObjectID, const double scale);
+  void setFrictionCoefficient(const int physObjectID,
+                              const double frictionCoefficient);
+  void setRestitutionCoefficient(const int physObjectID,
+                                 const double restitutionCoefficient);
+  void setLinearDamping(const int physObjectID, const double linDamping);
+  void setAngularDamping(const int physObjectID, const double angDamping);
+
+  // ============ Object Getter functions =============
+  const double getMass(const int physObjectID);
+  const Magnum::Vector3 getCOM(const int physObjectID);
+  const Magnum::Vector3 getInertiaVector(const int physObjectID);
+  const Magnum::Matrix3 getInertiaMatrix(const int physObjectID);
+  const double getScale(const int physObjectID);
+  const double getFrictionCoefficient(const int physObjectID);
+  const double getRestitutionCoefficient(const int physObjectID);
+  const double getLinearDamping(const int physObjectID);
+  const double getAngularDamping(const int physObjectID);
+
+  // ============= Platform dependent function =============
+  virtual const double getMargin(const int physObjectID) { return 0.0; };
+  virtual void setMargin(const int physObjectID, const double margin){};
+
+  // =========== Debug functions ===========
+  int checkActiveObjects();
+
+  //============ Interact with objects =============
+  // NOTE: engine specifics handled by objects themselves...
+  void applyForce(const int physObjectID,
+                  Magnum::Vector3& force,
```

Comment:
```suggestion
                  const Magnum::Vector3& force,
```

I believe this is an input only argument? i.e. `applyForce` does not change the value of `force`?